src/controllers/restatedeployment/reconcilers/knative.rs

```diff
@@ -0,0 +1,938 @@
+                    "Scheduling removal of old Configuration (after drain delay)"
+                );
+
+                let remove_at = chrono::Utc::now()
+                    .checked_add_signed(chrono::TimeDelta::minutes(5)) // Same as ReplicaSet cleanup
+                    .expect("remove_version_at in bounds");
+
+                let config_api: Api<Configuration> = Api::namespaced(ctx.client.clone(), namespace);
+                let params = PatchParams::apply("restate-operator/remove-version-at").force();
+
+                config_api
+                    .patch_metadata(
+                        &config_name,
+                        &params,
+                        &Patch::Apply(json!({
+                            "apiVersion": Configuration::api_version(&()),
+                            "kind": Configuration::kind(&()),
+                            "metadata": {
+                                "annotations": {
+                                    RESTATE_REMOVE_VERSION_AT_ANNOTATION: remove_at.to_rfc3339(),
+                                }
+                            }
+                        })),
+                    )
+                    .await?;
+
+                // ensure we keep track of the soonest remove_at
+                next_removal = match next_removal {
+                    None => Some(remove_at),
+                    Some(next_removal) if next_removal > remove_at => Some(remove_at),
+                    els => els,
+                };
+
+                continue;
+            }
+        }
+    }
+
+    Ok((active_count, next_removal))
+}
+
+/// Get tag from Configuration annotation
+fn get_configuration_tag(config: &Configuration) -> Option<String> {
+    config.metadata.annotations.as_ref()
+        .and_then(|a| a.get(RESTATE_TAG_ANNOTATION))
+        .cloned()
+}
+
+/// Delete Configuration and its associated Route
+async fn delete_configuration_and_route(
+    ctx: &Context,
+    namespace: &str,
+    config_name: &str,
+    rsd: &RestateDeployment,
+) -> Result<()> {
+    // Extract tag from configuration name to determine route name
+    // Configuration name format: {rsd-name}-{tag}
+    let rsd_name = rsd.name_any();
+    let route_name = if config_name.starts_with(&format!("{}-", rsd_name)) {
+        config_name.to_string()
+    } else {
+        // Fallback: assume same name
+        config_name.to_string()
+    };
+
+    info!(
+        configuration = %config_name,
+        namespace = %namespace,
+        "Deleting old Configuration"
+    );
+    let config_api: Api<Configuration> = Api::namespaced(ctx.client.clone(), namespace);
+    config_api.delete(config_name, &Default::default()).await?;
+
+    info!(
+        route = %route_name,
+        namespace = %namespace,
+        "Deleting old Route"
+    );
+    let route_api: Api<Route> = Api::namespaced(ctx.client.clone(), namespace);
+    // Route might not exist if it was already deleted, so ignore NotFound errors
```

Comment:
what else might have deleted it? 

also, there is a risk of a dangling route if we delete configuration, then panic and then we never get to this code again to delete the route. should we have one of them own the other, so deleting one deletes the other?